diff --git a/src/input/ButtonThread.cpp b/src/input/ButtonThread.cpp
index 251311df7..9119159c8 100644
--- a/src/input/ButtonThread.cpp
+++ b/src/input/ButtonThread.cpp
@@ -13,6 +13,7 @@
 #include "main.h"
 #include "modules/CannedMessageModule.h"
 #include "modules/ExternalNotificationModule.h"
+#include "modules/optional/CachedPhoneTracker/CachedPhoneTracker.h"
 #include "sleep.h"
 #ifdef ARCH_PORTDUINO
 #include "platform/portduino/PortduinoGlue.h"
@@ -255,17 +256,7 @@ int32_t ButtonThread::runOnce()
                 break;
 #if !HAS_SCREEN
             case 4:
-                if (moduleConfig.external_notification.enabled && externalNotificationModule) {
-                    externalNotificationModule->setMute(!externalNotificationModule->getMute());
-                    IF_SCREEN(if (!externalNotificationModule->getMute()) externalNotificationModule->stopNow();)
-                    if (externalNotificationModule->getMute()) {
-                        LOG_INFO("Temporarily Muted");
-                        play4ClickDown(); // Disable tone
-                    } else {
-                        LOG_INFO("Unmuted");
-                        play4ClickUp(); // Enable tone
-                    }
-                }
+                CachedPhoneTracker::toggleTrackerMode();
                 break;
 #endif
             // No valid multipress action
diff --git a/src/modules/Modules.cpp b/src/modules/Modules.cpp
index 1c32a0ab2..52ba92a95 100644
--- a/src/modules/Modules.cpp
+++ b/src/modules/Modules.cpp
@@ -41,6 +41,7 @@
 #endif
 #if !MESHTASTIC_EXCLUDE_GPS
 #include "modules/PositionModule.h"
+#include "modules/optional/CachedPhoneTracker/CachedPhoneTracker.h"
 #endif
 #if !MESHTASTIC_EXCLUDE_REMOTEHARDWARE
 #include "modules/RemoteHardwareModule.h"
@@ -162,6 +163,7 @@ void setupModules()
 #endif
 #if !MESHTASTIC_EXCLUDE_GPS
     positionModule = new PositionModule();
+    cachedPhoneTracker = new CachedPhoneTracker();
 #endif
 #if !MESHTASTIC_EXCLUDE_WAYPOINT
     waypointModule = new WaypointModule();
